### PR TITLE
Refactor: subscriptions

### PR DESCRIPTION
**Pull request by ricklue**
closing #536
All components with more than one subscription have now one general subscription where all subscribes are hooked to.

### CODE REVIEW

File: src/app/debug/filter-side-drawer/filter-side-drawer.component.ts [hunk 1]
```diff
@@ -52,35 +49,36 @@ export class FilterSideDrawerComponent implements OnDestroy, OnInit {
   }
 
   ngOnDestroy(): void {
-    this.unsubscribeAll();
+    this.genaralFilterSubscription?.unsubscribe();
```
Comment (MatthijsSmets) [inline src/app/debug/filter-side-drawer/filter-side-drawer.component.ts:52]: You can remove the ?

File: src/app/debug/filter-side-drawer/filter-side-drawer.component.ts [hunk 2]
```diff
@@ -52,35 +49,36 @@ export class FilterSideDrawerComponent implements OnDestroy, OnInit {
...
   }
 
   setSubscriptions(): void {
-    this.shouldShowFilterSubscription = this.filterService.showFilter$.subscribe({
-      next: (show: boolean) => (this.shouldShowFilter = show),
-      error: () => catchError(this.errorHandler.handleError()),
+    this.genaralFilterSubscription?.add(() => {
```
Comment (MatthijsSmets) [inline src/app/debug/filter-side-drawer/filter-side-drawer.component.ts:56]: Same here

File: src/app/debug/filter-side-drawer/filter-side-drawer.component.ts [hunk 3]
```diff
@@ -52,35 +49,36 @@ export class FilterSideDrawerComponent implements OnDestroy, OnInit {
...
+      this.filterService.showFilter$.subscribe({
+        next: (show: boolean) => (this.shouldShowFilter = show),
+        error: () => catchError(this.errorHandler.handleError()),
+      });
     });
-    this.metadataLabelsSubscription = this.filterService.metadataLabels$.subscribe({
-      next: (metadataLabels: string[]) => (this.metadataLabels = metadataLabels),
-      error: () => catchError(this.errorHandler.handleError()),
+    this.genaralFilterSubscription?.add(() => {
```
Comment (MatthijsSmets) [inline src/app/debug/filter-side-drawer/filter-side-drawer.component.ts:62]: Same here

**Review by MatthijsSmets — CHANGES_REQUESTED** [review]

File: src/app/debug/filter-side-drawer/filter-side-drawer.component.ts (hunk 1 above)
Comment (ricklue) [reply src/app/debug/filter-side-drawer/filter-side-drawer.component.ts]: I forgot to change it back, but now I have removed all the question marks

**Review by MatthijsSmets — APPROVED** [review]

**Review by MatthijsSmets — APPROVED** [review]

**Pull request merged**